src/attendees/attendee.py

```diff
@@ -79,7 +89,17 @@ def set_attribute(self, key: Hashable, value: Any,
             TypeError if key is not hashable.
             ValueError if key is "fname", "lname", or "email"
         """
+        self.key = key
+        self.value = value
+        self.verifyFunc = verifyFunc
+
         raise NotImplementedError
+        
+        if type(key) is not Hashable:
+            raise TypeError("key is not hashable")
+
+        if key is 'fname'
+            raise ValueError("key cannot be 'fname'")
```

Comment:
A better way of doing this is to use a list or tuple of invalid values, then use the `in` operator to see if `key` is one of these invalid values. THere's a better way of formatting the error message using f-strings.